### PR TITLE
CLN: reorganise vars in `Styler` init

### PR DESCRIPTION
**Pull request by attack68**
Trying to clean up Styler class grouping and ordering variables and adding typing where necessary.

### CODE REVIEW

**Comment by jreback** [comment]
sure can you merge master and ping on green.

**Comment by attack68** [comment]
@jreback merged and green

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -157,53 +157,40 @@ def __init__(
...
         if data.ndim == 1:
             data = data.to_frame()
         if not data.index.is_unique or not data.columns.is_unique:
             raise ValueError("style is not supported for non-unique indices.")
-
-        self.data = data
-        self.index = data.index
-        self.columns = data.columns
-
+        assert isinstance(data, DataFrame)
```
Comment (jreback) [inline pandas/io/formats/style.py:167]: this is superfluous, if you can remove in a followon (see L161)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @attack68

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:167]: this was added otherwsise mypy complains about mismatched type. maybe there is a better way..?

**Pull request merged**